### DISCUSSION TITLE
Requester state transitions

### DISCUSSION THREAD

**Discussion by steven-bellock** [DMTF/libspdm#720] [General]
Currently if the Requester receives an error response `SPDM_ERROR_CODE_REQUEST_RESYNCH` from the Responder libspdm automatically changes the connection state to `LIBSPDM_CONNECTION_STATE_NOT_STARTED`
https://github.com/DMTF/libspdm/blob/57ac200e7c0c8f633f1259a71d2f20c1c5f7b014/library/spdm_requester_lib/libspdm_req_handle_error_response.c#L99-L102
This violates a library pattern where state changes need to be initiated by the integrator. So in this specific case we could have it so that
1. The integrator would see that a resynch is requested and the libspdm connection state remains unchanged.
2. It is up to the integrator to call `libspdm_init_connection`.

There may be use cases (TBD) where the integrator doesn't want libspdm to automatically change state.

**Comment by jyao1** [comment]
3/21/2022 - we agree this is an issue. Need give a full list of improper state change.
